web-api/src/business/useCases/user/verifyUserPendingEmailInteractor.test.ts

```diff
@@ -0,0 +1,188 @@
+    ...validUser,
+    firstName: 'Olden',
+    lastName: 'Vivas',
+    pendingEmail: 'other@example.com',
+    pendingEmailVerificationToken: '42289629-abe1-46d7-b7a4-9d3834f919xd',
+    role: ROLES.petitioner,
+    userId: getContactPrimary(MOCK_CASE).contactId,
+  };
+
+  const mockCase = {
+    ...MOCK_CASE,
+    petitioners: [
+      {
+        ...getContactPrimary(MOCK_CASE),
+        serviceIndicator: SERVICE_INDICATOR_TYPES.SI_PAPER,
+      },
+    ],
+    privatePractitioners: [mockPractitioner],
+    status: CASE_STATUS_TYPES.generalDocket,
+  };
+
+  beforeEach(() => {
+    applicationContext.getCurrentUser.mockReturnValue(mockPractitioner);
+    applicationContext
+      .getPersistenceGateway()
+      .getUserById.mockReturnValue(mockPractitioner);
+
+    applicationContext
+      .getPersistenceGateway()
+      .isEmailAvailable.mockReturnValue(true);
+
+    applicationContext
+      .getPersistenceGateway()
+      .getCaseByDocketNumber.mockReturnValue(mockCase);
+
+    applicationContext
+      .getPersistenceGateway()
+      .getDocketNumbersByUser.mockReturnValue([mockCase.docketNumber]);
+  });
+
+  it('should throw unauthorized error when user does not have permission to verify emails', async () => {
+    applicationContext.getCurrentUser.mockReturnValue({
+      role: ROLES.petitionsClerk,
+      userId: 'f7d90c05-f6cd-442c-a168-202db587f16f',
+    });
+
+    await expect(
+      verifyUserPendingEmailInteractor(applicationContext, {
+        token: 'abc',
+      }),
+    ).rejects.toThrow('Unauthorized to manage emails');
+  });
+
+  it('should throw an unauthorized error when the token passed as an argument does not match stored token on user', async () => {
+    await expect(
+      verifyUserPendingEmailInteractor(applicationContext, {
+        token: 'abc',
+      }),
+    ).rejects.toThrow('Tokens do not match');
+  });
+
+  it('should throw an unauthorized error when the token passed as an argument and the token store on the user are both undefined', async () => {
+    applicationContext.getPersistenceGateway().getUserById.mockReturnValue({
+      ...mockPractitioner,
+      pendingEmailVerificationToken: undefined,
+    });
+
+    await expect(
+      verifyUserPendingEmailInteractor(applicationContext, {
+        token: undefined,
+      }),
+    ).rejects.toThrow('Tokens do not match');
+  });
+
+  it('should throw an error when the pendingEmail address is not available in cognito', async () => {
+    applicationContext
+      .getPersistenceGateway()
+      .isEmailAvailable.mockReturnValue(false);
+
+    await expect(
+      verifyUserPendingEmailInteractor(applicationContext, {
+        token: TOKEN,
+      }),
+    ).rejects.toThrow('Email is not available');
+  });
+
+  it('should update the cognito email when tokens match', async () => {
+    await expect(
+      verifyUserPendingEmailInteractor(applicationContext, {
+        token: TOKEN,
+      }),
+    ).resolves.not.toThrow('Tokens do not match');
+
```

Comment:
Perhaps remove the `'Tokens do not match'` string as I don't think this interactor should throw an error and it can be less restrictive?